server/src/termdb.server.init.ts

```diff
@@ -490,15 +490,15 @@ export function server_init_db_queries(ds) {
 		from which to derive forbiddenRoutes and clientAuthResults for ds to tailor chart support based on user role and permission
 	*/
 	q.getSupportedChartTypes = req => {
-		const { forbiddenRoutes, clientAuthResults } = authApi.getNonsensitiveInfo(req)
+		const authInfo = authApi.getNonsensitiveInfo(req)
 		const supportedChartTypes = {} // key: subcohort string, value: list of chart types allowed for this cohort
 
 		for (const [cohort, cohortTermTypes] of Object.entries(ds.cohort.termdb.termtypeByCohort.nested)) {
 			supportedChartTypes[cohort] = []
 
 			for (const [chartType, isSupported] of Object.entries(commonCharts)) {
 				// auth second argument may be ignored by isSupported() callback
-				if (isSupported({ ds, forbiddenRoutes, cohortTermTypes, clientAuthResults })) {
+				if (isSupported(Object.assign({ ds, cohortTermTypes }, authInfo))) {
```

Comment:
can use object srpead syntax, `{  ds, cohortTermTypes , ...authInfo }`, instead of Object.assign()